cluster/cluster.go

```diff
@@ -40,6 +45,77 @@ func Join(ctx context.Context, cfg Config) (*Cluster, error) {
 	}, nil
 }
 
+type MemberAddInfo struct {
+	InitialCluster      string
+	InitialClusterState string
+}
+
+func (c *Cluster) MemberAdd(ctx context.Context, name, peerURL string) (*MemberAddInfo, error) {
+	etcdCfg := c.etcd.Config()
+	cfg := clientv3.Config{
+		Endpoints:   []string{etcdCfg.LCUrls[0].String()},
+		DialTimeout: 5 * time.Second,
+	}
+	client, err := clientv3.New(cfg)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create etcd client from config: %w", err)
+	}
+
+	mresp, err := client.MemberAdd(ctx, []string{peerURL})
+	if err != nil {
+		return nil, fmt.Errorf("failed to add member with peerURL n%v: %w", peerURL, err)
+	}
+
+	initialClusterStrings := make([]string, 2)
+	for _, member := range mresp.Members {
+		if member.Name == "" && strings.Compare(peerURL, member.PeerURLs[0]) == 0 {
+			initialClusterStrings = append(initialClusterStrings, initialClusterStringFormatter(name, member.PeerURLs[0]))
+		} else {
+			initialClusterStrings = append(initialClusterStrings, initialClusterStringFormatter(member.Name, member.PeerURLs[0]))
+		}
+	}
+	sort.Strings(initialClusterStrings)
```

Comment:
What's the reason for sorting? For the output to be deterministic?

Reply:
Yeah, etcd seems like it returns a sorted one too when using their cli (not 100% sure). I'd rather not have a string such as `node7=_,node3=_,node1=_,node193=_,bob=_`, i'd imagine it'd be harder to debug any errors relating to the initial cluster if it were the case